python_src/src/lamp_py/performance_manager/l0_gtfs_static_mod.py

```diff
@@ -50,57 +51,40 @@ def generate_scheduled_travel_times(
     process_logger.log_complete()
 
 
-def generate_scheduled_branch_headways(
-    fk_timestamp: int, db_manager: DatabaseManager
-) -> None:
+def generate_scheduled_branch_headways(db_manager: DatabaseManager) -> None:
     """
     generate scheduled branch headways and insert into static_stop_times table
     """
-    branch_headways_cte = (
+    branch_headways = (
         sa.select(
-            StaticStopTimes.pk_id,
+            TempStaticHeadwaysGen.pk_id,
             (
-                StaticStopTimes.departure_time
-                - sa.func.lag(StaticStopTimes.departure_time).over(
+                TempStaticHeadwaysGen.departure_time
+                - sa.func.lag(TempStaticHeadwaysGen.departure_time).over(
                     partition_by=[
-                        StaticStops.parent_station,
-                        StaticTrips.service_id,
-                        StaticTrips.direction_id,
-                        StaticTrips.branch_route_id,
+                        TempStaticHeadwaysGen.parent_station,
+                        TempStaticHeadwaysGen.service_id,
+                        TempStaticHeadwaysGen.direction_id,
+                        TempStaticHeadwaysGen.branch_route_id,
                     ],
-                    order_by=StaticStopTimes.departure_time,
+                    order_by=TempStaticHeadwaysGen.departure_time,
                 )
             ).label("branch_headways"),
         )
-        .select_from(StaticStopTimes)
-        .join(
-            StaticTrips,
-            sa.and_(
-                StaticStopTimes.trip_id == StaticTrips.trip_id,
-                StaticStopTimes.timestamp == StaticTrips.timestamp,
-            ),
-        )
-        .join(
-            StaticStops,
-            sa.and_(
-                StaticStopTimes.stop_id == StaticStops.stop_id,
-                StaticStopTimes.timestamp == StaticStops.timestamp,
-            ),
-        )
+        .select_from(TempStaticHeadwaysGen)
         .where(
-            StaticStopTimes.timestamp == fk_timestamp,
-            StaticTrips.branch_route_id.is_not(None),
+            TempStaticHeadwaysGen.branch_route_id.is_not(None),
         )
-        .cte("scheduled_branch_headways")
+        .subquery("scheduled_branch_headways")
```

Comment:
what is the distinction between `cte` and `subquery` here?

Reply:
Very much like the which table to select from issue, it really doesn't matter. Traditionally the CTE let's you write much more human readable queries, with all of your CTE statement listed out before your final query. 

`subquery` will result in a set of nested select statements, the final result is basically un-readable. Since we're using an ORM type format to put all of this together, it really doesn't matter.